pipeline/src/aggregator/mod.rs

```diff
@@ -222,113 +235,94 @@ async fn process_feed_batch(ctx: SessionContext, conclusion_feed: DataFrame) ->
             "new_state",
         ])?
         .with_column_renamed("new_state", "state")?
-        // Write states to the doc_state table
-        .write_table("doc_state", DataFrameWriteOptions::new())
+        // Write states to the in memory doc_state table
+        .write_table(DOC_STATE_MEM_TABLE, DataFrameWriteOptions::new())
         .await
         .context("computing states")?;
+
+    let count = ctx.table(DOC_STATE_MEM_TABLE).await?.count().await?;
+    // If we have enough data cached in memory write it out to persistent store
+    if count >= max_cached_rows {
```

Comment:
I know I'm late to the party here, but is there an issue if no data gets written for a while that writes could sit in the memory table but never get flushed to disk?  Do we need a time-based flush condition in addition to a number of records based one?

Reply:
Its possible that writes stay in memory. However I do not think we need a time based flush mechanism because the data is derived and can be recomputed easily and quickly. 

If the process exists then any data in memory is lost. However on startup that data is recomputed and repopulated into memory. This has the nice behavior that we do not end up with small files in low write use cases. And since the buffer size is the same as the batch size this is means that the amount of data that needs to be recomputed on startup is always manageable, meaning is never larger than a batch we would have computed during normal operation anyways.

Reply:
okay cool.  So I guess the risk is just that the data Parquet in S3 is out of date during periods of low writes.  Better than data loss, if the latency becomes an issue for someone we can add a time-based flush then I suppose.

Reply:
Yep, and if they query the parquet data through the Flight SQL API on ceramc-one then the in memory data is used to serve the query